More docs (and RTD)

EntilZha opened the pull request:
With https://github.com/pytorch/text/pull/179 merged I went ahead and did a manual merge of changes I had from the docs I was working on. Mainly this turned out to be:

* Adding in Readthedocs specific stuff to get this working: https://torch-text.readthedocs.io/en/latest/
* Fixing some warnings from sphinx docs
* Adding a couple doc pages that weren't in the original PR.

The tentative plan is after merging that I'll change RTD to point to this repo instead of my fork.

jekbradbury commented:
Is there any chance the URL could be `torchtext.readthedocs.io` to align with the pip package name and `torchvision`?

EntilZha commented:
Done, now its at https://torchtext.readthedocs.io/en/latest/

I'll update the readme link in a sec.

jekbradbury commented on torchtext/data/field.py:
```diff
@@ -130,13 +130,11 @@ class Field(RawField):
+                 tensor_type=torch.LongTensor, preprocessing=None, postprocessing=None, lower=False,
+                 tokenize=(lambda s: s.split()), include_lengths=False, batch_first=False, pad_token="<pad>",
+                 unk_token="<unk>", pad_first=False):
+        super().__init__(preprocessing, postprocessing)
```
This line either shouldn't be present or should use the Python 2 `super(self, Field)` syntax.

EntilZha commented:
I removed that `super` line since its not relevant for docs, fixed some of the lint errors, and merged changes associated with moving from md to rst for the readme (badge). I think the PR should be good to go

EntilZha commented:
Last thing is that I changed the git project RTD monitors to this one. The last step would be to add the webhook on github which is readthedocs.org/api/v2/webhook/torchtext/22503/. Once that is done it should start auto-building the docs on each commit.

jekbradbury commented:
Added the webhook 👍; let me know if it's not working
